tests/unit/io/ARFFFile_unittest.cc

```diff
@@ -0,0 +1,258 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/io/ARFFFile.h>
+
+#include <sg_gtest_utilities.h>
+
+using namespace shogun;
+
+// Tolerance values for tests
+template <typename T>
+constexpr T get_epsilon()
+{
+	return std::numeric_limits<T>::epsilon() * 100;
+}
+template <>
+constexpr floatmax_t get_epsilon()
+{
+	return 1e-13;
+}
+
+// convert type to the supported enums
+template <typename T>
+constexpr EPrimitiveType convert_type_to_enum()
+{
+	return EPrimitiveType::PT_UNDEFINED;
+}
+
+template <>
+constexpr EPrimitiveType convert_type_to_enum<int8_t>()
+{
+	return EPrimitiveType::PT_INT8;
+}
+
+template <>
+constexpr EPrimitiveType convert_type_to_enum<int16_t>()
+{
+	return EPrimitiveType::PT_INT16;
+}
+
+template <>
+constexpr EPrimitiveType convert_type_to_enum<int32_t>()
+{
+	return EPrimitiveType::PT_INT32;
+}
+
+template <>
+constexpr EPrimitiveType convert_type_to_enum<int64_t>()
+{
+	return EPrimitiveType::PT_INT64;
+}
+
+template <>
+constexpr EPrimitiveType convert_type_to_enum<float32_t>()
+{
+	return EPrimitiveType::PT_FLOAT32;
+}
+
+template <>
+constexpr EPrimitiveType convert_type_to_enum<float64_t>()
+{
+	return EPrimitiveType::PT_FLOAT64;
+}
+
+template <>
+constexpr EPrimitiveType convert_type_to_enum<floatmax_t>()
+{
+	return EPrimitiveType::PT_FLOATMAX;
+}
```

Comment:
@karlnapf I added these utility functions might be worth putting them somewhere else?

Reply:
or it might already exist somewhere else?